Description: Check for sysfs' acpi file, procfs may not contain acpi data with recent kernels
Author: Andrey Danin <danindrey@mail.ru>, Eduard Bloch <blade@debian.org>
Bug: http://sourceforge.net/mailarchive/forum.php?thread_name=E1SHwYx-00029t-1N.danindrey-mail-ru%40f232.mail.ru&forum_name=icewm-devel
Forwarded: Yes

Index: icewm-1.3.7/src/wmtaskbar.cc
===================================================================
--- icewm-1.3.7.orig/src/wmtaskbar.cc	2010-10-31 15:09:36.000000000 +0100
+++ icewm-1.3.7/src/wmtaskbar.cc	2012-04-11 23:22:23.466564372 +0200
@@ -467,6 +467,7 @@
 #endif
 #ifdef CONFIG_APPLET_APM
     if (taskBarShowApm && (access(APMDEV, 0) == 0 ||
+                           access("/sys/class/power_supply", 0) == 0 ||
                            access("/proc/acpi", 0) == 0 ||
                            access("/dev/acpi", 0) == 0 ||
 	                   access("/proc/pmu", R_OK|X_OK) == 0))
